src/visualization/utilities.jl

```diff
@@ -859,6 +860,113 @@ function unstructured_3d_to_1d_curve(original_nodes, unstructured_data, nvisnode
   return arc_length, data_on_curve, nothing
 end
 
+# Convert 3d unstructured data from a general mesh to 1d data at given curve.
+function unstructured_3d_to_1d_curve(nodes, data, curve, slice, point, nvisnodes)
+    if curve === nothing
+        curve = axis_curve(nodes[1,:,:,:,:], nodes[2,:,:,:,:], nodes[3,:,:,:,:], slice, point, nvisnodes)
+    end
+
+    n_points_curve = size(curve, 2)
+    data_on_curve = Array{Float64}(undef, n_points_curve) #variables
+    for i in 1:n_points_curve
+        data_on_curve[i] = get_value_at_point(curve[:,i], nodes, data)
```

Comment:
Here (and below, e.g., in `is_valid_tetrahedron`) are many allocations. Does using `view`s help to improve performance?